internal/metadata/clustermdloader_test.go

```diff
@@ -49,6 +56,60 @@ func TestUpdatePlan(t *testing.T) {
 		}
 	})
 
+	t.Run("check if all clusters in constant were loaded", func(t *testing.T) {
+
+		type Teststruct struct {
+			//Kind string `yaml:"kind"`
+			Metadata struct {
+				Name string `yaml:"name"`
+			} `yaml:"metadata"`
+		}
+
+		p := plantypes.NewPlan()
+		loader := metadata.ClusterMDLoader{}
+		cmMap := loader.GetClusters(p)
+		//fmt.Println(cmMap)
+		files, err := ioutil.ReadDir("clusters/")
+		if err != nil {
+			fmt.Println("error reading clusters folder")
+		}
+
+		// filter the yaml files
+		re_yaml := regexp.MustCompile(".yaml")
+		re_yml := regexp.MustCompile(".yml")
+
+		var yfiles []string
+		for _, file := range files {
+			if re_yaml.MatchString(file.Name()) || re_yml.MatchString(file.Name()) {
+				yfiles = append(yfiles, file.Name())
+			}
+		}
+
+		// parse yaml files and collect the required info (medata-> names)
+		var metadata_names []string
+		fmt.Println("parsing yaml files:")
+		for _, yfile := range yfiles {
+			filename, _ := filepath.Abs("clusters/" + yfile)
+			//fmt.Println(filename)
+			content, _ := ioutil.ReadFile(filename)
+			test_struct := Teststruct{}
+			err = yaml.Unmarshal(content, &test_struct)
+			//fmt.Println(test_struct.Metadata.Name)
+			metadata_names = append(metadata_names, test_struct.Metadata.Name)
+		}
+		fmt.Println("finish parsing yaml files")
+		fmt.Println(metadata_names)
+
+		//TODO: Read all .yaml files in internal/metadata/clusters, and find the value in metadata.name using say a regex
+
+		for _, clustername := range metadata_names {
+			if _, ok := cmMap[clustername]; !ok {
```

Comment:
We should also possibly check the other way around, if there is a item in cmMap which is not in the metadata_names. It can even be two different tests.